tokio-tcp/src/listener.rs

```diff
@@ -178,6 +178,13 @@ impl TcpListener {
     /// # }
     /// ```
     pub fn poll_accept_std(&mut self) -> Poll<(net::TcpStream, SocketAddr), io::Error> {
+        // Error takes precedence, indicated by writable event on listening socket
```

Comment:
Is there a reason to put this first over trying to accept?

Reply:
I don't think so - I was partly thinking I could maybe leave the `try_ready!` logic in place, partly that if a writable flag does in fact indicate more severe problems then maybe we should deal with that first. But if I can work out the other issues I'd be happy to put read and write on equal footing.